### PR TITLE
optimize ci

### PR DESCRIPTION
**Pull request by consolethinks**
Optimize CI to only run configurations that have been affected by the changes

### CODE REVIEW

File: .github/workflows/compose_test.yaml [hunk 1]
```diff
@@ -48,19 +48,50 @@ jobs:
...
+              ${{ (contains(needs.get-changed-files.outputs.changed_files
+              , 'services/backend/services/v3')) &&
+              (! contains(needs.get-changed-files.outputs.changed_files
+              , 'services/backend/services/v4')) && 'v4' || ''}}
+          - ELASTIC_OR_JOBS_ENABLED:
+              ${{ (! contains(needs.get-changed-files.outputs.changed_files
+              , 'backend/services')) || 'nothing' }}
+          - LDAP_ENABLED:
+              ${{ (! contains(needs.get-changed-files.outputs.changed_files
+              , 'services/ldap')) || 'nothing' }}
```
Comment (fpotier) [inline .github/workflows/compose_test.yaml:91]: Would this set `LDAP_ENABLED` to `'nothing'` if there was no change? (Same for OIDC, elastic and jobs)

**Comment by fpotier** [comment]
I think this is a good start but I think the changed files rules don't account for every case.
For instance, from my understanding atm, changing the ldap config of the backend doesn't set `LDAP_ENABLED` to true.

File: .github/workflows/compose_test.yaml (hunk 1 above)
Comment (consolethinks) [reply .github/workflows/compose_test.yaml:108]: these are exceptions, so it will effectively try to remove any combinations made from the matrix that contains the `string` specified in the exclusion. So if the result is 'nothing', it will try to remove 'nothing' but as it doesn't exist, it won't remove anything. This was done in order to avoid removing the empty string ('') from the list of cases, because the default output of contains is 'true' (if it's true) or '' (if it's false)

**Comment by consolethinks** [comment]
> I think this is a good start but I believe the 'changed files' rules don't account for every case. For instance, from my understanding, changing the ldap config of the backend without changing anything in `services/ldap` would skip every matrix combination where `LDAP_ENABLED=true`

That is true, it would require a few extra `contains` expressions to check for this.

**Comment by sbliven** [comment]
I'm a bit concerned by the complexity introduced here. We can keep adding clauses to monitor more stuff, but at some point it becomes hard to maintain. Is speeding up the CI (#208) worth this, or would it be better to just run the full matrix?

**Comment by minottic** [comment]
I think this is a good change, the CI takes almost 10 minutes. So at least adding the if conditions I think it's necessary. The exclude block, as everything is almost there, adds more control over what gets run. The rules have to be reviewed, but it's not a major change, we just need to set some sensible "contains" for the configuration we want, and I think we can keep the list relatively short. 

TLDR; after reviewing the contains I would give this a try, and see how it works going forward, as it's almost ready for merge. If we are not happy, we'll simply remove the exclude block

**Comment by minottic** [comment]
as discussed offline with @consolethinks I think this is ready now. I would be happy to merge and give it a try. What do you think @fpotier , @sbliven ?

**Review by minottic — APPROVED** [review]

**Comment by fpotier** [comment]
> as discussed offline with @consolethinks I think this is ready now. I would be happy to merge and give it a try. What do you think @fpotier , @sbliven ?

I didn't review in detail the rules but I'm okay to give it a try

**Comment by minottic** [comment]
yes, these tests check if the components start fine with the correct dependencies. Integration tests are missing completely, that's probably a not too easy change
